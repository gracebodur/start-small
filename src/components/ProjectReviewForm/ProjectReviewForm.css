.ReviewForm select, .ReviewForm label {
    font-size: 1.1em;
  }
  
  .ReviewForm label {
    display: inline-block;
    margin: 10px 10px 20px;
    margin-left: 0;
  }
  
  .Button {
    display: flex;
    align-items: center;
    justify-content: center;
    background: #3D495A;
    margin-top: 5px;
    margin-bottom: 20px;
    border-radius: 5px;
    border: none;
    color: white;
    padding: 15px 32px;
    cursor: pointer;
  }

  .Post__review__btn {
    color: var(--clr-light);
    text-decoration: none;
    text-align: center;
    cursor: pointer;
    font-size: 16px;
    text-transform: uppercase;
    letter-spacing: 2px;
    font-family: var(--ff-primary);
  }

  .Stars__option {
    color: var(--clr-purple);
    font-family: var(--ff-primary);
  }

  .Text__area {
    color: var(--clr-purple);
    font-family: var(--ff-primary);
    font-size: 20px;
  }