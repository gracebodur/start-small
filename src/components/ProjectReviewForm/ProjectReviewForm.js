import React, { Component } from 'react'
import ProjectContext from '../../contexts/ProjectContext'
import ProjectsApiService from '../../services/projects-api-service'
import { Section, Button, Textarea } from '../Utils/Utils'
import './ProjectReviewForm.css'

export default class ReviewForm extends Component {
  static contextType = ProjectContext

  handleSubmit = ev => {
    ev.preventDefault()
    const { project } = this.context
    const { text, rating } = ev.target

    ProjectsApiService.postReview( project.project_id, text.value, Number( rating.value ) )
      .then( this.context.addReviews )
      .then( () => {
        text.value = ''
      } )
      .catch( this.context.setError )
  }

  render () {
    return (
      <Section>
      <form
        className='ReviewForm form-group'
        onSubmit={this.handleSubmit}
      >
      <label className='h5 review-label' htmlFor='rating'>How likely will you donate to this project?</label>
        <div className='text h4'>
          <Textarea
            className='Text__area'
            required
            aria-label='Type a review...'
            name='text'
            id='text'
            cols='25'
            rows='6'
            placeholder='Type a review..'>
          </Textarea>
        </div>

        <div className='select h3'>
          <select
            className='Stars__option'
            required
            aria-label='Rate this project!'
            name='rating'
            id='rating'
          >
            <option value='1'>1 Star</option>
            <option value='2'>2 Stars</option>
            <option value='3'>3 Stars</option>
            <option value='4'>4 Stars</option>
            <option value='5'>5 Stars</option>
          </select>
        </div>

        <Button type='submit' className='Post__review__btn'>
          Post review
        </Button>
      </form>
      </Section>
    )
  }
}

