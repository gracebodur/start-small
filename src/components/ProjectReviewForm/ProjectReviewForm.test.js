import React from "react";
import toJson from "enzyme-to-json";
import ProjectReviewForm from "./ProjectReviewForm";

describe(`ProjectReviewForm component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a ProjectReviewForm by default", () => {
    const wrapper = <ProjectReviewForm />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the ProjectReviewForm given props", () => {
    const wrapper = <ProjectReviewForm {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
