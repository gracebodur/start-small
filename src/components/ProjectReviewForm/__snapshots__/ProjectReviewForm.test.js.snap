// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectReviewForm component renders a ProjectReviewForm by default 1`] = `undefined`;

exports[`ProjectReviewForm component renders the ProjectReviewForm given props 1`] = `undefined`;
